docs/kubernetes_events.md

```diff
@@ -0,0 +1,89 @@
+* `ScheduledEvent`
+* `SQSTermination`
+* `SpotInterruption`
+
+Node action reasons:
+
+* `Cordon`
+* `CordonError`
+* `CordonAndDrain`
+* `CordonAndDrainError`
+* `PreDrain`
+* `PreDrainError`
+* `PostDrain`
+* `PostDrainError`
+* `Uncordon`
+* `UncordonError`
+* `MonitorError`
+
+## Default annotations
+
+If `emit-kubernetes-events` is enabled, AWS Node Termination Handler will automatically inject a set of annotations to each event it emits. Such annotations are gathered from the underlying host's IMDS endpoint and enrich each event with information about the host that emitted it.
+
+_**NOTE**: In Queue Processor mode, these annotations will reflect the node running NTH not the node receiving the events. See [Caveats](https://github.com/aws/aws-node-termination-handler/blob/main/docs/kubernetes_events.md#caveats) for more information._
+
+The default annotations are:
+
+Name | Example value
+--- | ---
+`account-id` | `123456789012` 
+`availability-zone` | `us-west-2a`
+`instance-id` | `i-abcdef12345678901`
+`instance-life-cycle` | `spot`
+`instance-type` | `m5.8xlarge`
+`local-hostname` | `ip-10-1-2-3.us-west-2.compute.internal`
+`local-ipv4` | `10.1.2.3`
+`public-hostname` | `my-example.host.net`
+`public-ipv4` | `42.42.42.42`
+`region` | `us-west-2`
+
+If `kubernetes-events-extra-annotations` are specified they will be appended to the above. In case of collision, the user-defined annotation wins.
+
+## How to get events
+
+All events are about Kubernetes `Node` objects so they belong in the `default` namespace. The event source is `aws-node-termination-handler`. From command line, use `kubectl` to get the events as follows:
+
+```sh
+kubectl get events --field-selector "source=aws-node-termination-handler"
+```
+
+To narrow down the search you can use multiple field selectors, like:
+
+```sh
+kubectl get events --field-selector "reason=SpotInterruption,involvedObject.name=ip-10-1-2-3.us-west-2.compute.internal"
+```
+
+Results can also be printed out in JSON or YAML format and piped to processors like `jq` or `yq`. Then, the above annotations can also be used for discovery and filtering.
+
+## Caveats
+
+### Default annotations in Queue Processor Mode
+
+Default annotations values are gathered from the IMDS endpoint local to the Node on which AWS Node Termination Handler runs. This is fine when running on IMDS Processor Mode since an AWS Node Termination Handler Pod will be deployed to all Nodes via a `DaemonSet` and each Node will emit all events related to itself with its own default annotations.
+
+However, when running in Queue Processor Mode AWS Node Termination Handler is deployed to a number of Nodes (1 replica by default) via a `Deployment`. In that case the default annotations values will be gathered from the Node(s) running AWS Node Termination Handler, and so the values in the default annotations stamped to all events will match those of the Node from which the event was emitted, not those of the Node of which the event is about.
```

Comment:
Are the annotations worth anything in this case? It seems both misleading and unhelpful to print values for a different node than the event actually affected. Perhaps we should just omit the default annotations when running in queue processor mode, thoughts?

Reply:
I'm trying to think if you would want to know the metadata of the node **running nth** in queue processor mode. If not, it looks like the host instance running NTH is included in the event here, so we could omit:

```
"source": {
                "component": "aws-node-termination-handler",
                "host": "ip-10-1-2-3.us-east-2.compute.internal"
            }
```

> It seems both misleading and unhelpful to print values for a different node than the event actually affected

Also agree, but the latest commit should address this. Now, the messaging shown in the even will include the instanceId being affected:

`"message": "Rebalance recommendation received. Instance **<instance-id>** will be cordoned at 2021-04-26T15:10:48Z \n",`

Reply:
User provided annotations will still be worthy (think of annotations like cluster name, cluster owner, environment, etc) but data coming from IMDS might not make much sense and I agree it can be confusing, hence the notes in the doc. Probably the right move is to omit annotations with data coming from IMDS when in SQS mode.

Reply:
Last commit should do it. Also updated the doc to reflect these changes.